mindtrace/database/mindtrace/database/backends/unified_odm_backend.py

```
@@ -0,0 +1,936 @@
+            
+        Raises:
+            RuntimeError: If the active backend is not recognized.
+            
+        Example:
+            .. code-block:: python
+            
+                backend_type = unified_backend.get_current_backend_type()
+                if backend_type == BackendType.MONGO:
+                    print("Using MongoDB backend")
+                elif backend_type == BackendType.REDIS:
+                    print("Using Redis backend")
+        """
+        active = self._get_active_backend()
+        if active == self.mongo_backend:
+            return BackendType.MONGO
+        elif active == self.redis_backend:
+            return BackendType.REDIS
+        else:
+            raise RuntimeError("Unknown active backend")
+    
+    def is_async(self) -> bool:
+        """
+        Check if the currently active backend operates asynchronously.
+        
+        Returns:
+            bool: True if the active backend is asynchronous, False otherwise.
+            
+        Example:
+            .. code-block:: python
+            
+                if unified_backend.is_async():
+                    result = await unified_backend.insert_async(document)
+                else:
+                    result = unified_backend.insert(document)
+        """
+        return self._get_active_backend().is_async()
+    
+    async def initialize_async(self):
+        """
+        Initialize asynchronous backends (MongoDB).
+        
+        This method should be called in an async context when using MongoDB
+        as it requires async initialization. Redis backends are initialized
+        synchronously and don't require this call.
+        
+        Example:
+            .. code-block:: python
+            
+                # In an async function
+                await unified_backend.initialize_async()
+        """
+        if self.mongo_backend:
+            await self.mongo_backend.initialize()
+    
```

Explicitly calling `await initialize_async()` when only a sync redis backend has been passed in `__init__()` should raise an appropriate error. Currently is a no-op and user does not know.

---

Does this [commit ](https://github.com/Mindtrace/mindtrace/pull/19/commits/044722844b365583024df1d6af9e2772c7163e46)address your concerns?
https://github.com/Mindtrace/mindtrace/pull/19/commits/044722844b365583024df1d6af9e2772c7163e46